vault/ui_custom_message.go

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/helper/namespace"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	UICustomMessageKey string = "/custom-messages"
+
+	MaximumCustomMessageCount int = 100
```

Comment:
Suggestion: rename this to `MaximumCustomMessageCountPerNamespace` -- from reading the test that seems to be what it does (right?) but reading this value at first, I assumed it was the global limit.